packages/core/src/app/payment/Payment.tsx

```diff
@@ -372,24 +372,25 @@ class Payment extends Component<
 
             if (isErrorWithType(error)) {
                 const { body, headers, status } = error;
-
-                if (body.type === 'provider_error' && headers.location) {
-                    window.top?.location.assign(headers.location);
-                }
-
-                // Reload the checkout object to get the latest `shouldExecuteSpamCheck` value,
-                // which will in turn make `SpamProtectionField` visible again.
-                // NOTE: As a temporary fix, we're checking the status code instead of the error
-                // type because of an issue with Nginx config, which causes the server to return
-                // HTML page instead of JSON response when there is a 429 error.
-                if (
-                    status === 429 ||
-                    body.type === 'spam_protection_expired' ||
-                    body.type === 'spam_protection_failed'
-                ) {
-                    this.setState({ didExceedSpamLimit: true });
-
-                    await loadCheckout();
+                if (body || status === 429){
```

Comment:
If `body` is `null` and `status` is 429, the line below `if (body.type === 'provider_error' && headers.location) {` will still break?

I think the quickest fix is to just put a question mark after `body`, like `body?.type` within this method.

Reply:
Agh good point, thanks for the advice. Looks like it was fixed in a separate PR but went ahead and made this change for practice.